packages/shared_preferences/shared_preferences_platform_interface/test/method_channel_shared_preferences_test.dart

```diff
@@ -44,9 +44,9 @@ void main() {
           return await testData.clear();
         }
         final RegExp setterRegExp = RegExp(r'set(.*)');
-        final Match match = setterRegExp.matchAsPrefix(methodCall.method);
-        if (match.groupCount == 1) {
-          final String valueType = match.group(1);
+        final Match? match = setterRegExp.matchAsPrefix(methodCall.method);
+        if (match != null && match.groupCount == 1) {
```

Comment:
Nit: couldn't this be `if (match?.groupCount == 1)` (since null != 1)?

Reply:
Ah, good suggestion. done

Reply:
But we are already checking if match != null in the first condition. I believe it won’t evaluate the second statement if match is null